community/kernel/src/main/java/org/neo4j/kernel/impl/util/Neo4jJobScheduler.java

```diff
@@ -67,11 +76,29 @@ public Executor executor( final Group group )
     }
 
     @Override
+    public ExecutorService workStealingExecutor( Group group, int parallelism )
+    {
+        return workStealingExecutors.computeIfAbsent( group, g -> createNewWorkStealingExecutor( g, parallelism ) );
+    }
+
+    @Override
     public ThreadFactory threadFactory( final Group group )
     {
         return job -> createNewThread( group, job, NO_METADATA );
     }
 
+    private ExecutorService createNewWorkStealingExecutor( Group group, int parallelism )
+    {
+        final ForkJoinPool.ForkJoinWorkerThreadFactory factory = pool ->
+        {
+            final ForkJoinWorkerThread worker = ForkJoinPool.defaultForkJoinWorkerThreadFactory.newThread( pool );
+            worker.setName( group.threadName( new HashMap<>() ) );
+            return worker;
+        };
+
+        return new ForkJoinPool( parallelism, factory, null, false );
```

Comment:
There's a ForkJoinPool constructor that takes a worker name prefix. I think that'll be simpler than building this custom thread factory.

Reply:
Oh wait, that constructor is private.